server/services/overviewPageService.ts

```diff
@@ -118,7 +133,7 @@ export default class OverviewPageService {
       prisonerDetail,
     ] = await Promise.all([
       prisonApiClient.getStaffRoles(staffId, userCaseLoads.find(caseload => caseload.currentlyActive)?.caseLoadId),
-      curiousApiClient.getLearnerNeurodivergence(prisonerData.prisonerNumber),
+      Result.wrap(getLearnerNeurodivergence, pageErrorCallback)(),
```

Comment:
When `Promise.all` resolves, `wrap` surrounds the function in a try catch to enable it to return a `Result`.  This avoids the need to turn this into a `Promise.allSettled` and then handle everything immediately.